main.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional
+
+from fastapi import FastAPI, Depends, HTTPException, Query
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(
+    db: Session = Depends(get_db),
+    skip: int = 0,
+    limit: int = 10,
+):
+    return crud.get_all_author(db=db, skip=skip, limit=limit)
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(author: schemas.AuthorCreate, db: Session = Depends(get_db)):
+    db_author = crud.get_author_by_name(db=db, name=author.name)
+    if db_author:
+        raise HTTPException(
+            status_code=400, detail=f"Author with name {author.name} already exists"
+        )
+
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/books/", response_model=list[schemas.Book])
+def read_books(
+    db: Session = Depends(get_db),
+    skip: int = Query(0, ge=0),
+    limit: int = Query(10, ge=1),
+    author: Optional[str] = None,
+):
+    return crud.get_book_list(db=db, skip=skip, limit=limit, author=author)
+
+
+@app.get("/books/{book_id}/", response_model=schemas.Book)
+def read_single_book(
+    book_id: int,
```

Comment:
The `book_id` parameter in the `read_single_book` function is correctly typed as `int`, which is good for ensuring the correct data type is passed.